plutus-pab/src/Plutus/PAB/Webserver/API.hs

```diff
@@ -43,8 +44,8 @@ type API t walletId -- see note [WalletID type in wallet API]
                         :<|> "stop"     :> Description "Terminate the instance." :> Put '[JSON] ()
                         )
                     )
-            :<|> "instances" :> "wallet" :> Capture "wallet-id" walletId :> Description "List of all active contract instances for the wallet." :>  Get '[JSON] [ContractInstanceClientState t]
-            :<|> "instances" :> Description "List of all active contract instances." :> Get '[JSON] [ContractInstanceClientState t]
+            :<|> "instances" :> "wallet" :> Capture "wallet-id" walletId :> QueryParam "status" Text :> Description "List of contract instances for the wallet filtered by status (active, stopped, done). All by default." :>  Get '[JSON] [ContractInstanceClientState t]
```

Comment:
This looks fine; but is it possible with servant to do this in a type-safe way?

Reply:
Initially I was going to use `ContractActivityStatus` defined above, but then I remembered about the issue with the purescript generator as it has problems with types there. That's why we use `Capture "contract-instance-id" Text`. Maybe it's only for newtypes and `data` types are fine. Not sure, worth to test.

Reply:
Yes it would be worth testing it, but I am afraid that the purescript bridge won't support it.